subsys/net/lib/sockets/sockets.c

```diff
@@ -45,8 +45,36 @@ const struct socket_op_vtable sock_fd_op_vtable;
 static inline void *get_sock_vtable(
 			int sock, const struct socket_op_vtable **vtable)
 {
-	return z_get_fd_obj_and_vtable(sock,
-				       (const struct fd_op_vtable **)vtable);
+	void *ctx;
+
+	ctx = z_get_fd_obj_and_vtable(sock,
+				      (const struct fd_op_vtable **)vtable);
+
+#ifdef CONFIG_USERSPACE
+	if (z_is_in_user_syscall()) {
+		struct z_object *zo;
+		int ret;
+
+		zo = z_object_find(ctx);
+		ret = z_object_validate(zo, K_OBJ_NET_SOCKET, _OBJ_INIT_TRUE);
+
+		if (ret != 0) {
+			z_dump_object_error(ret, ctx, zo, K_OBJ_NET_SOCKET);
+			/* Invalidate the context, the caller doesn't have
+			 * sufficient permission or there was some other
+			 * problem with the net context object
```

Comment:
Probably better to say here `problem with the net socket object`

Reply:
done.